src/parser.ts

```diff
@@ -792,7 +803,7 @@ export async function parseCustomConfigProvider(cancel: vscode.CancellationToken
     let numberOfLines: number = dryRunOutputLines.length;
     let index: number = 0;
     let done: boolean = false;
-    function doParsingChunk(): void {
+    async function doParsingChunk(): Promise<void> {
```

Comment:
I think we need a new version of `scheduleTask` that deals with awaitable functions.  `scheduleAsyncTask` maybe?